Formula/caffe.rb

```diff
@@ -0,0 +1,55 @@
+class Caffe < Formula
+  desc "Fast open framework for deep learning"
+  homepage "https://caffe.berkeleyvision.org/"
+  url "https://github.com/BVLC/caffe/archive/1.0.tar.gz"
+  sha256 "71d3c9eb8a183150f965a465824d01fe82826c22505f7aa314f700ace03fa77f"
+
+  depends_on "cmake" => :build
+  depends_on "boost"
+  depends_on "gflags"
+  depends_on "glog"
+  depends_on "hdf5"
+  depends_on "leveldb" => :optional
```

Comment:
Deps should be alphabetised with `:build` options being first, and `:optional` options being last here. The one deviation from the normal rule is that `depends_on "snappy" if build.with? "leveldb"` should be the last dep here.